libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -105,6 +105,36 @@ void CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element)
                 return;
             }
         }
+
+    } else if (("cpu.shares" == settingKey)) {
+        long int newValue;
+        try {
+            newValue = std::stol(settingValue);
+        } catch (std::out_of_range &err) {
+            std::string errorMessage = "The value for cpu.shares is too high";
+            log_debug() << errorMessage;
+            throw InvalidValueError(errorMessage);
+        } catch (std::invalid_argument &err) {
+            std::string errorMessage = "The value for cpu.shares is not an integer";
+            log_debug() << errorMessage;
+            throw InvalidValueError(errorMessage);
+        }
+
+        if (newValue < 2) {
+            std::string errorMessage = "Value of cpu.shares must be at least 2";
+            log_debug() << errorMessage;
+            throw InvalidValueError(errorMessage);
+        }
+        // if the new value is smaller/equal than the old value, then we don't save the new value.
+        if (m_settings.count(settingKey) != 0) {
+            if (std::stol(m_settings[settingKey]) >= newValue) {
```

Comment:
If choosing integer, then I guess `std::stoi` should be used here later on as well. Btw, I think the check should be the other way: if the `newValue` is larger, we save it. Giving a smaller value isn't even worthy of a debug output imho :)